compiler/erlang_check.erl

```diff
@@ -245,29 +244,26 @@ check_module(File) ->
             report,
             % By adding debug_info, we ensure that the output of xref:m will
             % contain the caller MFAs too.
-            debug_info,
-            {i, AbsDir ++ "/include"},
-            {i, AbsDir ++ "/../include"},
-            {i, AbsDir ++ "/../../include"},
-            {i, AbsDir ++ "/../../../include"}],
-
-    case process_rebar_configs(AbsDir) of
-        {ok, RebarOpts} ->
-            code:add_patha(absname(AbsDir, "ebin")),
-            {CompileOpts0, OutDir} =
-                case get(outdir) of
-                    undefined ->
-                        % strong_validation = we only want validation, don't
-                        % generate beam file
-                        {[strong_validation], undefined};
-                    OutDir0 ->
-                        AbsOutDir = filename:join(AbsDir, OutDir0),
-                        {[{outdir, AbsOutDir}], AbsOutDir}
-                end,
-            CompileOpts = CompileOpts0 ++ Defs ++ RebarOpts,
+            debug_info],
+
+    BuildSystemOpts = case guess_build_system(AbsDir) of
+                          {rebar, RebarFiles} ->
+                              load_rebar_files(RebarFiles);
+                          {make, Makefiles} ->
+                              load_makefiles(Makefiles)
+                      end,
```

Comment:
When a project doesn't use either rebar or make, we still want to try compiling the Erlang module. (Imagine that you simply create a small Erlang file and want to compile it.)

This `case` doesn't handle that scenario, and nor does the `guess_build_system` function:

    $ .../vim-erlang-compiler/compiler/erlang_check.erl x.erl
    escript: exception error: {function_clause,
                         [{local,guess_build_system,["/home/hcs/__/.",[]]}]}